src/Features/CSharp/Portable/SplitOrMergeIfStatements/CSharpIfLikeStatementGenerator.cs

```diff
@@ -0,0 +1,151 @@
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.SplitOrMergeIfStatements;
+
+namespace Microsoft.CodeAnalysis.CSharp.SplitOrMergeIfStatements
+{
+    [ExportLanguageService(typeof(IIfLikeStatementGenerator), LanguageNames.CSharp), Shared]
+    internal sealed class CSharpIfLikeStatementGenerator : IIfLikeStatementGenerator
+    {
+        public bool IsIfLikeStatement(SyntaxNode node) => node is IfStatementSyntax;
+
+        public bool IsCondition(SyntaxNode expression, out SyntaxNode ifLikeStatement)
+        {
+            if (expression.Parent is IfStatementSyntax ifStatement && ifStatement.Condition == expression)
+            {
+                ifLikeStatement = ifStatement;
+                return true;
+            }
+
+            ifLikeStatement = null;
+            return false;
+        }
+
+        public bool IsElseIfClause(SyntaxNode node, out SyntaxNode parentIfLikeStatement)
+        {
+            if (node is IfStatementSyntax && node.Parent is ElseClauseSyntax)
+            {
+                parentIfLikeStatement = (IfStatementSyntax)node.Parent.Parent;
+                return true;
+            }
+
+            parentIfLikeStatement = null;
+            return false;
+        }
+
+        public SyntaxNode GetCondition(SyntaxNode ifLikeStatement)
+        {
+            var ifStatement = (IfStatementSyntax)ifLikeStatement;
+            return ifStatement.Condition;
+        }
+
+        public SyntaxNode GetRootIfStatement(SyntaxNode ifLikeStatement)
+        {
+            var ifStatement = (IfStatementSyntax)ifLikeStatement;
+
+            while (ifStatement.Parent is ElseClauseSyntax elseClause)
+            {
+                ifStatement = (IfStatementSyntax)elseClause.Parent;
+            }
+
+            return ifStatement;
+        }
+
+        public ImmutableArray<SyntaxNode> GetElseLikeClauses(SyntaxNode ifLikeStatement)
+        {
+            var ifStatement = (IfStatementSyntax)ifLikeStatement;
+
+            var builder = ImmutableArray.CreateBuilder<SyntaxNode>();
+
+            while (ifStatement.Else?.Statement is IfStatementSyntax elseIfStatement)
+            {
+                builder.Add(elseIfStatement);
+                ifStatement = elseIfStatement;
+            }
+
+            if (ifStatement.Else != null)
+            {
+                builder.Add(ifStatement.Else);
+            }
+
+            return builder.ToImmutable();
+        }
+
+        public SyntaxNode WithCondition(SyntaxNode ifLikeStatement, SyntaxNode condition)
+        {
+            var ifStatement = (IfStatementSyntax)ifLikeStatement;
+            return ifStatement.WithCondition((ExpressionSyntax)condition);
+        }
+
+        public SyntaxNode WithStatement(SyntaxNode ifLikeStatement, SyntaxNode statement)
+        {
+            var ifStatement = (IfStatementSyntax)ifLikeStatement;
+            return ifStatement.WithStatement(SyntaxFactory.Block((StatementSyntax)statement));
```

Comment:
interesting that this wraps with Block.

Reply:
➡️ Renamed in 8b346dcfe115a0adab33843a13abd305434d13f4